### PR TITLE
Add `Obj` functions for accessing the scannable prefix len of a mixed block

### PR DESCRIPTION
**Pull request by ncik-roberts**
This PR makes it possible to update generic OCaml traversals to also work for mixed blocks.

I add enough of an API for people to use this without allocating if they wish.

### CODE REVIEW

**Comment by ncik-roberts** [comment]
This isn't quite ready for review as it doesn't handle bytecode correctly. Will reopen soon.

**Comment by ncik-roberts** [comment]
It's now ready for review. I nominate @ccasin. There are runtime changes, but they don't require that much context.

**Comment by mshinwell** [comment]
I can look at the runtime changes.

File: ocaml/stdlib/obj.mli [hunk 1]
```diff
@@ -172,3 +172,35 @@ module Ephemeron: sig
   (** Maximum length of an ephemeron, ie the maximum number of keys an
       ephemeron could contain *)
 end
+
+module Uniform_or_mixed : sig
+  (** Blocks with a nominally scannable tag can still have a suffix of
+      unscanned objects; such a block is "mixed". This contrasts with
+      "uniform" blocks which are either all-scanned or all-unscanned.
+  *)
```
Comment (ccasin) [inline ocaml/stdlib/obj.mli:180]: Maybe it's worth noting somewhere in here that these functions can return different results on nativecode and bytecode, which might be a surprising source of test failures to someone who hasn't thought about the bytecode representation?

**Review by ccasin — APPROVED** [review]
Looks good.  One small suggestion, which you can feel free to ignore if you disagree.

File: ocaml/stdlib/obj.mli (hunk 1 above)
Comment (ncik-roberts) [reply ocaml/stdlib/obj.mli:184]: Added a comment about this, thanks.

**Pull request merged**